### ISSUE TITLE
Open multiple instances

### ISSUE THREAD

**Issue by Randallflagg0** [rgriebl/brickstore#370]
An important feature I have found with brickstock is the ability to open another instance to allow me to compare 2 files or work with multiple monitors rather than click back and forth within the program.  If the restriction to only open 1 instance could be lifted that would be great.

**Comment by ZZJHONS** [comment]
I would like it too, but just not by default, I mean, in the old BrickStock if you had it open and doble clicked one file to open it, it would open another instance instead inside of the already running on

**Comment by Randallflagg0** [comment]
@ZZJHONS I actually prefer it that way.  Easer to mouseover the task bar icon to see which files you have open than to click though all the files in 1 program.

**Comment by rgriebl** [comment]
This is not an arbitrary restriction:
1. the disk cache would get out of sync
2. the in-memory cache size would be doubled (tripled, etc) and with the high-res pictures nowadays, the cache size can easily go over 2GB
3. the solution would be to have one BrickStore process, but have  the possibility to move tabs/documents to a new window. I'd like to have that as well, but this needs some substantial refactoring of the way tabs/documents are managed internally.

TL;DR: it's on my to-do list, but the problem is much harder to solve than it may look.

**Comment by rgriebl** [comment]
Current dev versions come with the possibility to split the main window horizontally and vertically. I'll add a way to detach documents into separate windows, after more people have tested the splitting feature and I can be sure that it's bug free.

**Comment by rgriebl** [comment]
Adding split-off windows was easy enough with the new view architecture in place. Let's see how many bug reports come in :)

You can now split the main window horizontally and vertically (plus you can repeat this process recursively) and you can split-off separate windows (these do not have menubars or toolbars by design, but keyboard shortcuts and the context menu work).

If you are really desperate, you can start BrickStore.exe with the `--new-instance` parameter and you will get a second BrickStore process, even if there is one running already. This comes with all the disadvantages mentioned above, so it's not something that I would recommend or support in case of problems.

**Comment by ZZJHONS** [comment]
@rgriebl now if you already have Brickstore open and double click a file in the explorer it makes a new instance, I ended up with 3 running until I noticed as it opens the same file of the original first instance in the others.

**Comment by rgriebl** [comment]
Doesn't happen for me. Did you start any of these process with `--new-instance`?

**Comment by ZZJHONS** [comment]
Mmm no, but now I can't reproduce it, maybe was a little bug somewhere. Will reply again if I can reproduce.

**Comment by ZZJHONS** [comment]
<img width="840" alt="image" src="https://user-images.githubusercontent.com/1063368/151656437-baea3475-7465-47fb-9d53-3a44e7c13f8f.png">

Found out, just clicking twice multiple bsx files in a Windows folder.

**Comment by rgriebl** [comment]
Was one of the instances running before clicking twice? If yes: which one crashed: the original one or one of the newly started? And when did it crash: Immediately or after doing something in the UI?
I think there's a potential race-condition when starting too often in parallel, but the crash is worrying me.

**Comment by ZZJHONS** [comment]
You should have received a few sentry reports...

I can replicate the opening of multiple instances either with BS initially running or not, and it doesn't always crashes 1 instance, sometimes up to 7 or so work and none crashes and other times 2 or more work but just up to 1 crashes.
